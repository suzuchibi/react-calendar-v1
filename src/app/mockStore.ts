import configureMockStore from 'redux-mock-store';
const mock = configureMockStore();

export const mockStore = mock({
  drawer: false,
  today: 1626015600000,
  modal: {
    isActive: false,
    mode: '0',
  },
  form: {
    id: '0',
    isDuplicate: false,
  },
  cal: {
    current: 1625065200000,
    holidays: [
      {
        date: '2021-07-22',
        summary: '海の日',
      },
    ],
    status: 'loaded',
  },
  calPoper: {
    anchorEl: null,
  },
  schedule: {
    holidays: [],
    selected: 1626015600000,
    status: 'loaded',
  },
  datas: [
    {
      id: '1',
      title: 'タイトル1',
      coment: 'コメント1',
      start: 1626139200000,
      end: 1626141600000,
    },
  ],
  screen: false,
  snack: {
    open: false,
    msg: '',
  },
});
