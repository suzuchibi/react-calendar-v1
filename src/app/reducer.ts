import { combineReducers } from 'redux';
import isDrawerReducer from '../features/side/isDrawerReducer';
import todayReducer from '../features/today/todayReducer';
import modalReducer from '../features/modal/reducer';
import formReducer from '../features/form/reducer';
import calendarReducer from '../features/calendar/calendarReducer';
import calendarPoperReducer from '../features/dialog/calendarPoper/reducer';
import scheduleReducer from '../features/scheduler/reducers/scheduleReducer';
import scheduleDatasReducer from '../features/scheduler/reducers/datasReducer';
import fullScreenDialogReducer from '../features/dialog/fullScreen/reducer';
import snackReducer from '../features/snack/snackReducer';

const rootReducer = combineReducers({
  drawer: isDrawerReducer,
  today: todayReducer,
  modal: modalReducer,
  form: formReducer,
  cal: calendarReducer,
  calPoper: calendarPoperReducer,
  schedule: scheduleReducer,
  datas: scheduleDatasReducer,
  screen: fullScreenDialogReducer,
  snack: snackReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export default rootReducer;
