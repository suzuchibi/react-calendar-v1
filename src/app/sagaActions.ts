import { Holidays } from '../features/calendar/types';

const GOOGLE_CALENDAR_API = 'app/googleCalendarAPI';
const GOOGLE_CALENDAR_API_SUCESS = 'app/googleCalendarApiSuccess';
const GOOGLE_CALENDAR_API_ERROR = 'app/googleCalendarApiError';

export interface Props {
  mode: string;
  stamp: number;
}

export interface GoogleCalendarApiProps {
  type: typeof GOOGLE_CALENDAR_API;
  payload: Props;
}

interface HolidayProps {
  mode: string;
  result: Array<Holidays>;
}

export interface GoogleCalendarApiSuccessProps {
  type: typeof GOOGLE_CALENDAR_API_SUCESS;
  payload: HolidayProps;
}

export interface GoogleCalendarApiErrorProps {
  type: typeof GOOGLE_CALENDAR_API_ERROR;
  payload: {
    mode: string;
  };
}

export type ActionTypes =
  | GoogleCalendarApiProps
  | GoogleCalendarApiSuccessProps
  | GoogleCalendarApiErrorProps;

/**
 * google calendar API saga start
 * @param {object} payload
 */
export const gApiSaga = (payload: Props): ActionTypes => ({
  type: GOOGLE_CALENDAR_API,
  payload,
});

/**
 * Success
 * @param {object} payload
 */
export const gApiSuccess = (payload: HolidayProps): ActionTypes => ({
  type: GOOGLE_CALENDAR_API_SUCESS,
  payload,
});

/**
 * Error
 * @param {object} payload
 */
export const gApiError = (payload: { mode: string }): ActionTypes => ({
  type: GOOGLE_CALENDAR_API_ERROR,
  payload,
});
