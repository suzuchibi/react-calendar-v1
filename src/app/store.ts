import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import rootSaga from './sagas';
import rootReducer from './reducer';

const saga = createSagaMiddleware();
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(saga))
);
saga.run(rootSaga);

export default store;
