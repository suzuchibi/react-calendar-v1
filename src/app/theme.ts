import { createMuiTheme } from '@material-ui/core/styles';

export const theme = createMuiTheme({
  palette: {
    background: {
      default: '#fff',
    },
  },
});
