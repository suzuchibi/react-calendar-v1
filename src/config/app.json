{
  "drawerWidth": 260,
  "timeLineWidth": 60,
  "startOrder": 10,
  "lastOrder": 19,
  "dayOfTheWeek": ["日", "月", "火", "水", "木", "金", "土"],
  "gapiCal": {
    "key": "AIzaSyCLqi5tQwWw8qEGAWc9NC1cEU78C4EAWkM",
    "address": "ja.japanese#holiday@group.v.calendar.google.com",
    "gurl": "https://www.googleapis.com/calendar/v3/calendars/"
  }
}