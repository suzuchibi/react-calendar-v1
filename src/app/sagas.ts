import { put, call, takeEvery } from 'redux-saga/effects';
import { CallEffect, PutEffect } from 'redux-saga/effects';
import { Props as SagaProps } from './sagaActions';
import { Holidays } from '../features/calendar/types';
import GAPI from '../api/googleCalendarApi';

import config from '../config/app.json';
import {
  getYear,
  getMonth,
  getDate,
  startOfMonth,
  endOfMonth,
  add,
} from 'date-fns';

export interface ApiProps {
  type: string;
  payload: SagaProps;
}
export interface SuccessProps {
  type: string;
  payload: {
    mode: string;
    result: Array<Holidays>;
  };
}
export interface ErrorProps {
  type: string;
  payload: {
    mode: string;
  };
}
export type SagaTypes = CallEffect | PutEffect;

/**
 * SetDateText タイムゾーンのテキスト化
 * @param {Date | number} e
 * @return {String}
 */
const setTimeText = (e: Date | number) => {
  const pad = (t: number) => ('00' + t).slice(-2);
  return `${getYear(e)}-${pad(getMonth(e) + 1)}-${pad(getDate(e))}T00:00:00Z`;
};

/**
 * GetTimeZone 開始と終了の日付取得
 * @param {number} stamp
 * @param {String} mode
 * @return {Object<{start: String, end: String}>}
 */
const getTimeZone = (stamp: number, mode: String) => {
  switch (mode) {
    case 'cal':
      const s = startOfMonth(stamp);
      const e = endOfMonth(stamp);
      return {
        start: setTimeText(s),
        end: setTimeText(e),
      };
    case 'schedule':
      const dateAddSix = add(stamp, { days: 7 });
      return {
        start: setTimeText(stamp),
        end: setTimeText(dateAddSix),
      };
    case 'all':
      const sAll = startOfMonth(stamp);
      const end = endOfMonth(stamp);
      const endAddSix = add(end, { days: 6 });
      return {
        start: setTimeText(sAll),
        end: setTimeText(endAddSix),
      };
    default:
      return {
        start: '',
        end: '',
      };
  }
};

// ここからジェネレーター

/**
 * Success 成功した後のディスパッチ処理
 * @param {object} payload
 * @returns {void}
 */
export function* success({
  payload,
}: SuccessProps): Generator<PutEffect, void, SuccessProps> {
  const { mode, result } = payload;
  switch (mode) {
    case 'cal':
      yield put({ type: 'calendar/changeHolidays', payload: result });
      yield put({ type: 'calendar/changeStatus', payload: 'loaded' });
      break;
    case 'schedule':
      yield put({ type: 'schedule/changeHolidays', payload: result });
      yield put({ type: 'schedule/changeStatus', payload: 'loaded' });
      break;
    case 'all':
      yield put({ type: 'calendar/changeHolidays', payload: result });
      yield put({ type: 'schedule/changeHolidays', payload: result });
      yield put({ type: 'calendar/changeStatus', payload: 'loaded' });
      yield put({ type: 'schedule/changeStatus', payload: 'loaded' });
      break;
  }
}

/**
 * Error 失敗した後のディスパッチ処理
 * @param {object} payload modeの値が返る
 * @returns {void}
 */
export function* error({
  payload,
}: ErrorProps): Generator<PutEffect, void, string> {
  const { mode } = payload;
  switch (mode) {
    case 'cal':
      yield put({ type: 'calendar/changeHolidays', payload: [] });
      yield put({ type: 'calendar/changeStatus', payload: 'loaded' });
      break;
    case 'schedule':
      yield put({ type: 'schedule/changeHolidays', payload: [] });
      yield put({ type: 'schedule/changeStatus', payload: 'loaded' });
      break;
    case 'all':
      yield put({ type: 'calendar/changeHolidays', payload: [] });
      yield put({ type: 'schedule/changeHolidays', payload: [] });
      yield put({ type: 'calendar/changeStatus', payload: 'loaded' });
      yield put({ type: 'schedule/changeStatus', payload: 'loaded' });
      break;
  }
}

/**
 * Generator GoogleCalendarApi
 * @param {object} payload
 * @returns {void}
 */
export function* googleCalendarApiSaga({
  payload,
}: ApiProps): Generator<SagaTypes, void, any> {
  const { mode, stamp } = payload;
  const { start, end } = getTimeZone(stamp, mode);
  const param = `timeMin=${start}&timeMax=${end}&singleEvents=true`;
  const encode = encodeURIComponent(config.gapiCal.address);
  const path = `${config.gapiCal.gurl}${encode}/events?${param}`;
  try {
    // GoogleCalendarAPIコール
    const result = yield call(GAPI.start, path);
    yield put({
      type: 'app/googleCalendarApiSuccess',
      payload: { mode: mode, result: result },
    });
  } catch (err) {
    yield put({ type: 'snac/changeMsg', payload: err });
    yield put({ type: 'snac/changeOpen', payload: true });
    yield put({
      type: 'app/googleCalendarApiError',
      payload: { mode: mode },
    });
  }
}

export default function* rootSaga() {
  yield takeEvery('app/googleCalendarAPI', googleCalendarApiSaga);
  yield takeEvery('app/googleCalendarApiSuccess', success);
  yield takeEvery('app/googleCalendarApiError', error);
}
