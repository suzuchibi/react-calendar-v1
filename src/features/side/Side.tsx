import React, { FC } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import classNames from 'classnames';
import config from '../../config/app.json';
import { RootState } from '../../app/reducer';
import { makeStyles } from '@material-ui/core/styles';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import { Drawer, List, Divider } from '@material-ui/core';
import Calendar from '../calendar/compornents/Calendar';
import ListSubheader from '@material-ui/core/ListSubheader';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
import green from '@material-ui/core/colors/green';

// Actions
import { handleID } from '../form/actions';
import { handleIsActive, handleMode } from '../modal/actions';
import { handleDrawer } from './isDrawerReducer';

const drawerWidth = config.drawerWidth;

/**
 * CSS Styling
 * @return {Object}
 */
const useStyles = makeStyles(() => ({
  drawerWrapper: {
    position: 'relative',
  },
  sideTitle: {
    lineHeight: '1.8em',
    paddingTop: '4px',
  },
  listWrapper: {
    width: drawerWidth,
  },
  centerMargin: {
    display: 'flex',
    margin: '0 auto',
  },
  addButton: {
    color: '#fff',
    background: green[600],
    '&:hover': {
      background: green[800],
    },
  },
}));

/**
 * Render
 */
const Side: FC = () => {
  const classes = useStyles();
  const isDrawer = useSelector((state: RootState) => state.drawer);
  const md = useMediaQuery('(max-width:960px)');
  const sm = useMediaQuery('(max-width:600px)');

  const dispatch = useDispatch();
  const close = () => dispatch(handleDrawer(!isDrawer));
  const handleModal = () =>
    dispatch(handleMode('0')) &&
    dispatch(handleID('0')) &&
    dispatch(handleIsActive(true));
  const variantMode = (flag: Boolean) => {
    return flag ? 'temporary' : 'persistent';
  };
  const isDrawerDevice = () => {
    return sm ? false : !isDrawer;
  };

  return (
    <Drawer
      className={classes.drawerWrapper}
      variant={variantMode(md)}
      anchor="left"
      open={isDrawerDevice()}
      onClose={close}
    >
      <List className={classes.listWrapper}>
        <ListSubheader className={classes.sideTitle}>Calender</ListSubheader>
        <ListItem>
          <Calendar md={md} />
        </ListItem>
        <Divider />
        <ListItem>
          <ListItemText>
            <Button
              aria-label="add"
              variant="contained"
              className={classNames([classes.centerMargin, classes.addButton])}
              startIcon={<AddIcon />}
              disableElevation
              onClick={handleModal}
            >
              新規登録
            </Button>
          </ListItemText>
        </ListItem>
      </List>
    </Drawer>
  );
};

export default Side;
