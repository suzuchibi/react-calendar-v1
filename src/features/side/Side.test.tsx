import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Side from './Side';

import { mockStore as store } from '../../app/mockStore';

// Termial
// yarn test src/features/side/Side.test.ts

describe('<Side>', () => {
  it('スナップショット テスト', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Side />
        </MemoryRouter>
      </Provider>
    );
    // snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
