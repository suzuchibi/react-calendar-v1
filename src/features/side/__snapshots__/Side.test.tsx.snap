// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Side> スナップショット テスト 1`] = `
<DocumentFragment>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawerWrapper-1"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <ul
        class="MuiList-root makeStyles-listWrapper-3 MuiList-padding"
      >
        <li
          class="MuiListSubheader-root makeStyles-sideTitle-2 MuiListSubheader-sticky MuiListSubheader-gutters"
        >
          Calender
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters"
        >
          <div
            class="Calendar-cal-6"
          >
            <div
              class="TitleAndPrevNext-calNav-8"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                2021 
                <span
                  class="TitleAndPrevNext-calNavJweek-9"
                >
                  (令和3年)
                </span>
                 / 07
              </p>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="Calendar-calBox-7"
            >
              <ul
                class="dayOfTheWeek-calWeek-10"
              >
                <li
                  class="dayOfTheWeek-calWeekList-11 dayOfTheWeek-sunday-12"
                >
                  日
                </li>
                <li
                  class="dayOfTheWeek-calWeekList-11"
                >
                  月
                </li>
                <li
                  class="dayOfTheWeek-calWeekList-11"
                >
                  火
                </li>
                <li
                  class="dayOfTheWeek-calWeekList-11"
                >
                  水
                </li>
                <li
                  class="dayOfTheWeek-calWeekList-11"
                >
                  木
                </li>
                <li
                  class="dayOfTheWeek-calWeekList-11"
                >
                  金
                </li>
                <li
                  class="dayOfTheWeek-calWeekList-11 dayOfTheWeek-saturday-13"
                >
                  土
                </li>
              </ul>
              <div
                class="CalendarBoard-calBox-15"
              >
                <ul
                  class="CalendarBoard-calWeek-16"
                >
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-glayText-24"
                  >
                    27
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-glayText-24"
                  >
                    28
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-glayText-24"
                  >
                    29
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-glayText-24"
                  >
                    30
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    1
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    2
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-saturday-21"
                  >
                    3
                  </li>
                </ul>
                <ul
                  class="CalendarBoard-calWeek-16"
                >
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-sunday-22"
                  >
                    4
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    5
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    6
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    7
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    8
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    9
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-saturday-21"
                  >
                    10
                  </li>
                </ul>
                <ul
                  class="CalendarBoard-calWeek-16"
                >
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-sunday-22"
                  >
                    11
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-today-19 CalendarBoard-selected-20"
                  >
                    12
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    13
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    14
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    15
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    16
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-saturday-21"
                  >
                    17
                  </li>
                </ul>
                <ul
                  class="CalendarBoard-calWeek-16"
                >
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-sunday-22"
                  >
                    18
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    19
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    20
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    21
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-holiday-23"
                  >
                    22
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    23
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-saturday-21"
                  >
                    24
                  </li>
                </ul>
                <ul
                  class="CalendarBoard-calWeek-16"
                >
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-sunday-22"
                  >
                    25
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    26
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    27
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    28
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    29
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18"
                  >
                    30
                  </li>
                  <li
                    class="CalendarBoard-calWeekList-17 CalendarBoard-calLink-18 CalendarBoard-saturday-21"
                  >
                    31
                  </li>
                </ul>
                <ul
                  class="CalendarBoard-calWeek-16"
                />
              </div>
            </div>
          </div>
        </li>
        <hr
          class="MuiDivider-root"
        />
        <li
          class="MuiListItem-root MuiListItem-gutters"
        >
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              <button
                aria-label="add"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-centerMargin-4 makeStyles-addButton-5 MuiButton-disableElevation"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                  新規登録
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
